Formula/vis.rb

```diff
@@ -0,0 +1,51 @@
+class Vis < Formula
+  desc "Vim-like text editor"
+  homepage "https://github.com/martanne/vis"
+  url "https://github.com/martanne/vis/archive/v0.4.tar.gz"
+  sha256 "f11ba41cfb86dd39475960abfd12469de4da0ccfdb941f1d7680d89d987694c5"
+  head "https://github.com/martanne/vis.git"
+
+  depends_on "lua"
+  depends_on "libtermkey"
+
+  resource "lpeg" do
+    url "https://luarocks.org/manifests/gvvaughan/lpeg-1.0.1-1.src.rock", :using => :nounzip
+    sha256 "149be31e0155c4694f77ea7264d9b398dd134eca0d00ff03358d91a6cfb2ea9d"
+  end
+
+  def install
+    luapath = libexec/"vendor"
+    ENV["LUA_PATH"] = "#{luapath}/share/lua/5.2/?.lua"
+    ENV["LUA_CPATH"] = "#{luapath}/lib/lua/5.2/?.so"
+
+    resources.each do |r|
+      r.stage do
+        system "luarocks", "build", r.name, "--tree=#{luapath}"
+      end
+    end
+
+    # avoid inclusion of -lncursesw alongside -ltermkey because lncursesw might not exist (also avoid use of lncursesw altogether to simplify things)
+    inreplace "configure" do |s|
+      s.gsub!("for libcurses in ncursesw ncurses libcurses; do", "for libcurses in ncurses libcurses; do")
+      s.gsub!("LDFLAGS_TERMKEY=\"-ltermkey -lncursesw\"", "LDFLAGS_TERMKEY=\"-ltermkey\"")
+    end
+
+    system "./configure", "--prefix=#{libexec}"
+    system "make", "install"
+    (bin/"vise").write <<~EOS
+      #!/bin/sh
+      VIS_BASE=#{libexec}
+      VIS_PATH=$VIS_BASE/share/vis $VIS_BASE/bin/vis $@
+    EOS
+  end
+
+  def caveats; <<~EOS
+    To avoid a name conflict with the OS X system utility /usr/bin/vis,
+    this text editor must be invoked by calling `vise` ("vis-editor").
+  EOS
+  end
+
+  test do
+    assert_match "vis v0.4 +curses +lua", shell_output("#{bin}/vise -v 2>&1")
```

Comment:
And use `#{version}` here instead of hardcoding the version manually, although ILZ may well ask for a fuller functionality test if one is possible.